### PR TITLE
Improve REPL SIGINT user experience.

### PR DESCRIPTION
**Pull request by yuyichao**
- Add a SIGINT dead time after a force throw
  
  Fixes Jeff's issue in #17706
- Make the eval and print loop sigatomic to avoid sigint being delivered
  outside `try`-`catch` blocks.

### CODE REVIEW

**Comment by JeffBezanson** [comment]
:100: Massive improvement.

File: base/REPL.jl [hunk 1]
```diff
@@ -74,6 +76,7 @@ function eval_user_input(ast::ANY, backend::REPLBackend)
             iserr, lasterr = true, (err, catch_backtrace())
         end
     end
+    ccall(:jl_sigatomic_end, Void, ())
```
Comment (JeffBezanson) [inline base/REPL.jl]: I don't see why this is needed --- isn't it dominated by a begin-end sequence?

Comment (yuyichao) [inline base/REPL.jl]: It isn't really needed in the sense that it should be unreachable. It is needed if we exit normally from the loop though so I included it for completeness (admittedly it won't help if one return from the loop).

It isn't dominated by a `begin`-`end` since the `leave` from the `try`-`catch` implicitly reset the defer_signal counter.

**Comment by StefanKarpinski** [comment]
Can you summarize the behavioral upshot of this change?

**Comment by JeffBezanson** [comment]
You get a prompt back after ^C instead of julia exiting.

**Comment by StefanKarpinski** [comment]
Brillant.
